lib/liquid/block.rb

```diff
@@ -13,10 +20,22 @@ def parse(tokens)
       @blank = true
       @nodelist ||= []
       @nodelist.clear
+      @@whitespace ||= false
 
       while token = tokens.shift
         case token
         when IsTag
+          @@whitespace = false
+          if token =~ IsWhiteSpaceTagEnd
+            @@whitespace = true
```

Comment:
I think this flag should be moved to the tokenizer so that it applies to the next token.  It should also be named more appropriately to reflect that it will trim whitespace rather than saying the token is whitespace.